tests/integration/helpers.py

```diff
@@ -379,7 +381,8 @@ def _configure_dockerhub_mirror(conn: SSHConnection, dockerhub_mirror: str | Non
     if not dockerhub_mirror:
         return
     command = f'sudo mkdir -p /etc/docker/ && \
-echo {{ \\"registry-mirrors\\": [\\"{dockerhub_mirror}\\"]}} | sudo tee /etc/docker/daemon.json'
+echo {{ \\"registry-mirrors\\": [\\"{dockerhub_mirror.geturl()}\\"]}} | sudo tee \
```

Comment:
I think the URL validation should be the scope in which the application takes - whether the service is reachable depends on the SLA of the service and we shouldn't error out on the application during the URL check because the state of the URL can diverge during build time and URL reach check time. WDYT?